test/SDiagonal.jl

```diff
@@ -0,0 +1,104 @@
+@testset "SDiagonal" begin
+    @testset "Constructors" begin
+        @test SDiagonal{1,Int64}((1,)).diag === SVector{1,Int64}((1,))
+        @test SDiagonal{1,Float64}((1,)).diag === SVector{1,Float64}((1,))
+     
+        @test SDiagonal{4,Float64}((1, 1.0, 1, 1)).diag.data === (1.0, 1.0, 1.0, 1.0)
+        @test SDiagonal{4}((1, 1.0, 1, 1)).diag.data === (1.0, 1.0, 1.0, 1.0)
+        @test SDiagonal((1, 1.0, 1, 1)).diag.data === (1.0, 1.0, 1.0, 1.0)
+
+        # Bad input
+        @test_throws Exception SMatrix{1,Int}()
+        @test_throws Exception SMatrix{2,Int}((1,))
```

Comment:
`@test_throws Exception` is generally a very bad idea. It will pass if the code has a typo.